src/post/post.controller.ts

```diff
@@ -46,13 +46,29 @@ export class PostController {
     return this.postService.findOne(+id);
   }
 
+  @ApiOperation({ summary: 'Update a post'})
+  @ApiTags('post')
   @Patch(':id')
-  update(@Param('id') id: string, @Body() updatePostDto: UpdatePostDto) {
-    return this.postService.update(+id, updatePostDto);
+  async update(@Param('id') id: string, @Body() updatePostDto: UpdatePostDto, @Res() res) {
+    try {
+      const existingPost = await this.postService.update(id, updatePostDto);
```

Comment:
this variable name should be updatingPost

Reply:
It's ok, but I need to check a post exist or not, so i think existingPost still right. And if you want me to change, I will change